poc/policy/histogram_with_aggregator_dp.py

```diff
@@ -0,0 +1,82 @@
+"""
+A policy that uses only Aggregator-DP to achieve central (eps, delta)-DP.
+
+Each Aggregator independently adds discrete Gaussian noise to its aggregate
+share, so that as long as at least one Aggregator is honest, the final
+aggregate result is (eps, delta)-differentially private.
+"""
+
+# Access files that live in different directories, since currently the source
+# files are not module-based.
+import os
+import sys
+# `dir_name` points to the `poc` folder in DP draft.
+dir_name = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+# Access files in `mechanism` folder.
+sys.path.append(os.path.join(dir_name, "mechanism"))
+# Access files in VDAF `poc` folder.
+sys.path.append(os.path.join(dir_name, "draft-irtf-cfrg-vdaf", "poc"))
+from flp_generic import Histogram
+
+from discrete_gaussian import DiscreteGaussianWithZeroMean
+from policy import DpPolicy
+
+
+class HistogramWithAggregatorDp(DpPolicy):
+    Field = Histogram.Field
+    # A measurement is an unsigned integer, indicating an index
+    # less than `Histogram.length`.
+    Measurement = Histogram.Measurement
+    AggShare = list[Field]
+    AggResult = Histogram.AggResult
+    # The final aggregate result should be a vector of *signed* integers,
+    # because discrete Gaussian could produce negative noise which may have
+    # a larger absolute value than the count before noise.
+    DebiasedAggResult = list[int]
+
+    def __init__(self, epsilon: float, delta: float):
+        # TODO(junyechen1996): Compute discrete Gaussian parameter to
+        # achieve (eps, delta)-DP based on
+        # https://github.com/BorjaBalle/analytic-gaussian-mechanism.
+        self.dgauss_mechanism = DiscreteGaussianWithZeroMean(100.0)
+
+    def add_noise_to_measurement(self,
+                                 meas: Measurement,
+                                 ) -> Measurement:
+        # Nothing to do for Client-DP.
+        return meas
+
+    def add_noise_to_agg_share(self,
+                               agg_share: AggShare,
+                               meas_count: int,
+                               min_batch_size: int,
+                               ) -> AggShare:
+        """
+        Sample discrete Gaussian noise, and merge it with
+        aggregate share.
+        """
+        raise NotImplementedError()
+
+    def debias_agg_result(self,
+                          agg_result: AggResult,
+                          meas_count: int,
+                          min_batch_size: int,
+                          num_aggregators: int,
+                          ) -> DebiasedAggResult:
+        # TODO(junyechen1996): Interpret large unsigned integers as negative
```

Comment:
For this particular reason, this API's name may need to be changed to accommodate this problem. It's not strictly limited to debiasing for Client-DP. We basically need a way to "postprocess" the aggregate result from VDAF, knowing the DP noise we added.

Reply:
Good catch, let's generalize in this PR? Just call it "postprocess" for now and consider listing "debiasing the noise" as one possible application.

Reply:
I think this still counts as debias. Postprocess is too vague a term, it's hard to explain what should and shouldn't go into postprocess. So here you basically need to convert a List[Field] to list[int] first, then add noise?